### PR TITLE
Refactoring: Modularized panels and virtual panel

### PR DESCRIPTION
**Pull request by kgutwin**
## Description

This is one of two parts of a planned refactoring series as discussed in #61. This part consists of:

- Separating out the I2C and GPIO-based "Micro Panel" interface into its own class
- Separating out the display timer logic into its own class
- Adding a "virtual panel" which currently shows up as another tab in the OctoPrint main UI
- Adding support for multiple panels in parallel, as well as allowing the prime panel implementation to determine the dimensions of the panel
- Trimming down the settings logic, pushing most of it into the panel classes to do their own updates as needed

This PR is carefully designed to be independent of, and complementary to, the second PR in the refactoring series. It's recommended that both are merged at the same time, but it is not necessary.

## Checklist

- [x] Commit message describes the changes
- [x] Updated CHANGELOG.md "Unreleased" section with changes

### CODE REVIEW

File: octoprint_display_panel/templates/display_panel_tab.jinja2
Comment (sethvoltz) [inline octoprint_display_panel/templates/display_panel_tab.jinja2:8]: Ha, fantastic. This is exactly what I had in mind if I ever got to this.

**Review by sethvoltz — COMMENTED** [review]
This looks great. A few comments in here for you but if you don't get to them or make any changes, I plan to approve and merge later tonight my time.

**Comment by sethvoltz** [comment]
Just saw that this was also a WIP PR. Holding off on approving until you promote it to a full PR

File: octoprint_display_panel/__init__.py
Comment (kgutwin) [reply octoprint_display_panel/__init__.py:22]: I think I'll take these comments out before the PR is approved, no need to put them in just to take them out later.

File: octoprint_display_panel/__init__.py
Comment (kgutwin) [reply octoprint_display_panel/__init__.py]: Reverted to the original version. Eventually it may make sense to replace all the calls to `log_error()` with `self._logger.exception(message)` which is a nice way to log the full traceback.

**Comment by sethvoltz** [comment]
I just merged the other PR and wanted to give you a chance to fix the conflicts here before I start reviewing again

**Comment by sethvoltz** [comment]
@kgutwin checking in if you wouldn't mind fixing the merge conflicts and I'll review and merge.

**Comment by kgutwin** [comment]
Thanks for the reminder, I did a quick pass through to resolve the conflicts. Give it a try now!

**Review by sethvoltz — APPROVED** [review]
Thanks again for doing this!

**Pull request merged**
